### PR TITLE
[feature](compact) Duplicate with no keys tables compaction coredump

### LINKED ISSUE #18621: [Enhancement] duplicate table without key
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Doris skips data by short index, it is resource consuming to sorting data. Sometimes users do not need short index at all, so we can cancel enfore sort property on duplicate table.

### Solution

- ddl which creates a table without key
- load
- compaction
- query

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by herry2038**
# Proposed changes

Issue Number: close #18621

## Problem summary

1. Duplicate with no keys tables when compaction will coredump;
2.Vitical compaction will build heap table to do compaction, it will be no need for duplicate no keys tables.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

### CODE REVIEW

**Comment by herry2038** [comment]
run buildall

**Comment by herry2038** [comment]
run buildall

**Comment by herry2038** [comment]
run buildall

**Comment by herry2038** [comment]
run buildall

**Comment by herry2038** [comment]
run compile

**Comment by dataroaring** [comment]
run buildall

File: be/src/vec/olap/vertical_merge_iterator.cpp [hunk 1]
```diff
@@ -504,6 +504,103 @@ Status VerticalHeapMergeIterator::init(const StorageReadOptions& opts) {
...
+        }
+
+        RETURN_IF_ERROR(ctx->advance());
+        if (!ctx->valid()) {
+            _cur_iter_ctx = nullptr;
+            // push next iterator in same rowset into heap
+            auto cur_order = ctx->order();
+            while (cur_order + 1 < _iterator_init_flags.size()) {
+                auto& next_iter = _origin_iters[cur_order + 1];
+                VerticalMergeIteratorContext* next_ctx = new VerticalMergeIteratorContext(
```
Comment (dataroaring) [inline be/src/vec/olap/vertical_merge_iterator.cpp:542]: please use unique_ptr

File: be/src/vec/olap/vertical_merge_iterator.h [hunk 2]
```diff
@@ -299,6 +299,58 @@ class VerticalHeapMergeIterator : public RowwiseIterator {
...
+private:
+    // It will be released after '_merge_heap' has been built.
+    std::vector<RowwiseIteratorUPtr> _origin_iters;
+    std::vector<bool> _iterator_init_flags;
+    std::vector<RowsetId> _rowset_ids;
+    size_t _ori_return_cols;
+
+    const Schema* _schema = nullptr;
+
+    VerticalMergeIteratorContext* _cur_iter_ctx;
```
Comment (dataroaring) [inline be/src/vec/olap/vertical_merge_iterator.h:343]: please use unique_ptr

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by herry2038** [comment]
run buildall

**Pull request merged**